daceml/onnx/op_implementations/img_op_implementations.py

```diff
@@ -0,0 +1,360 @@
+        new_sdfg.arrays["Y"].transient = False
+
+        # add init state
+        # yapf: disable
+        init_state.add_mapped_tasklet("init",
+                                      map_ranges={
+                                          "i{}".format(i): "0:{}".format(s)
+                                          for i, s in enumerate(Y.shape)
+                                      },
+                                      inputs={},
+                                      code="y = {}".format(dtypes.min_value(Y.dtype)),
+                                      outputs=dict(
+                                          y=dace.Memlet("Y[{}]".format(
+                                              ", ".join("i{}".format(i)
+                                                        for i, _ in enumerate(Y.shape))))
+                                      ),
+                                      external_edges=True)
+        # yapf: enable
+
+        # the outer map loops over every entry in the output array
+        outer_me, outer_mx = new_state.add_map(
+            'outer_conv_map',
+            dict(b="0:{}".format(batch_size),
+                 c="0:{}".format(num_channels),
+                 out_x="0:{}".format(output_size_x),
+                 out_y="0:{}".format(output_size_y)))
+
+        # the inner map computes the value for a single entry in the output array (i.e. Y[b, c, x, y])
+        inner_me, inner_mx = new_state.add_map(
+            'inner_conv_map',
+            dict(hx="0:{}".format(filter_hx), hy="0:{}".format(filter_hy)))
+
+        compute_tasklet = new_state.add_tasklet("compute_entry",
+                                                inputs={"image_in"},
+                                                outputs={"output"},
+                                                code="output = image_in")
+
+        x_idx = _2d_sliding_window_index_expr(x_or_y="x",
+                                              stride=stride_x,
+                                              kernel_size=filter_hx)
+        y_idx = _2d_sliding_window_index_expr(x_or_y="y",
+                                              stride=stride_y,
+                                              kernel_size=filter_hy)
+
+        image_memlet = dace.Memlet("X[b, c, {}, {}]".format(x_idx, y_idx))
+
+        new_state.add_edge(inner_me, None, compute_tasklet, "image_in",
+                           image_memlet)
+
+        # hook up X
+        read_X = new_state.add_read("X")
+        inner_image_memlet = propagation.propagate_memlet(
+            new_state, image_memlet, inner_me, False)
+        outer_image_memlet = propagation.propagate_memlet(
+            new_state, inner_image_memlet, outer_me, False)
+        new_state.add_edge(outer_me, None, inner_me, None, inner_image_memlet)
+        new_state.add_edge(read_X, None, outer_me, None, outer_image_memlet)
+
+        # hook up outputs
+        output_memlet = dace.Memlet("Y[b, c, out_x, out_y]",
```

Comment:
hopefully this should just work (I've seen cases where it doesn't validate with cudnn because of "what happens if two values are equal" in the backprop)